src/cache/local.ml

```diff
@@ -388,32 +422,34 @@ let make ?(root = default_root ())
     ?(duplication_mode = detect_duplication_mode root)
     ?(log = Dune_util.Log.info) ?(warn = fun pp -> User_warning.emit pp)
     ~command_handler () =
-  if Path.basename root <> "v2" then
-    Result.Error "unable to read dune-cache"
-  else
-    let res =
-      { root
-      ; build_root = None
-      ; info = log
-      ; warn
-      ; repositories = []
-      ; command_handler
-      ; duplication_mode
-      ; temp_dir =
-          Path.temp_dir ~temp_dir:root "promoting."
-            ("." ^ string_of_int (Unix.getpid ()))
-      }
-    in
-    Path.mkdir_p @@ root_metadata res;
-    Path.mkdir_p @@ root_data res;
-    Result.ok res
+  let res =
+    { root
+    ; build_root = None
+    ; info = log
+    ; warn
+    ; repositories = []
+    ; command_handler
+    ; duplication_mode
+    ; temp_dir =
+        Path.temp_dir ~temp_dir:root "promoting."
+          ("." ^ string_of_int (Unix.getpid ()))
```

Comment:
It would be nice to introduce `val getpid : unit -> t` in `pid.ml` so that we don't use the untyped version of pid anywhere.

Reply:
Agreed. I've added a CR for myself, will do this in a separate PR.